common/datavalues/src/arrays/ops/if.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+
+use common_arrow::arrow::array::ArrayRef;
+use common_arrow::arrow::compute::if_then_else;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::prelude::*;
+
+macro_rules! impl_if_common {
+    ($lhs:ident, $rhs:ident,$predicate:ident) => {{
+        match ($predicate.len(), $lhs.len(), $rhs.len()) {
```

Comment:
Why not keep the order as macros define.

Reply:
fix